api/src/glfs-fops.c

```diff
@@ -5092,6 +5085,7 @@ glfs_realpath_common(struct glfs *fs, const char *path, char *resolved_path,
                      gf_boolean_t warn_deprecated)
 {
     int ret = -1;
+    xlator_t *this = THIS;
```

Comment:
`THIS` will probably be incorrect before `__GLFS_ENTRY_VALIDATE_FD()`.

Please check other places where this also happens.

Reply:
done